### PR TITLE
8289572: InputStream wrapping with BufferedInputStream is redundant in HttpTimestamper

### PR DESCRIPTION
**Pull request by stsypanov**
In case we read all bytes from an `InputStream` we don't need wrapping with `BufferedInputStream` as the bytes are not written into internal buffer. With removal of redundant buffer we save 8 kB of allocated memory.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8289572](https://bugs.openjdk.org/browse/JDK-8289572): InputStream wrapping with BufferedInputStream is redundant in HttpTimestamper

### Reviewers
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)
 * [Sean Mullan](https://openjdk.org/census#mullan) (@seanjmullan - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9343/head:pull/9343` \
`$ git checkout pull/9343`

Update a local copy of the PR: \
`$ git checkout pull/9343` \
`$ git pull https://git.openjdk.org/jdk pull/9343/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9343`

View PR using the GUI difftool: \
`$ git pr show -t 9343`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9343.diff">https://git.openjdk.org/jdk/pull/9343.diff</a>

</details>

### CODE REVIEW

**Comment by dfuch** [comment]
> In case we read all bytes from an InputStream we don't need wrapping with BufferedInputStream as the bytes are not written into internal buffer. With removal of redundant buffer we save 8 kB of allocated memory.

I believe that in that case the bytes might get double buffered, depending on whether the buffer used by `readAllBytes` is bigger than the buffer used by  `BufferedInputStream`, since IIAMNM `readAllBytes()` will call `read(byte[], off, len)` - but I otherwise agree that regardless of that wrapping in a `BufferedInputStream` is always wasteful in this case. Good find!

**Comment by stsypanov** [comment]
@dfuch as of current implementation of `InputStream.readNBytes()` it relies on `InputStream.DEFAULT_BUFFER_SIZE` which is 8192 and equals to `DEFAULT_BUFFER_SIZE` in `BufferedInputStream`.

**Review by jaikiran — APPROVED** [review]
Looks fine to me, but please wait for someone from the security team to review this, before merging.

**Review by seanjmullan — APPROVED** [review]

**Comment by stsypanov** [comment]
@seanjmullan done

**Comment by jaikiran** [comment]
I see that Sean has approved this PR. I'll run some tests internally with this change and if it goes fine, then I'll go ahead and sponsor this today.

**Comment by jaikiran** [comment]
The tests completed without any related issues.

**Pull request closed without merging**
